client/src/hooks/graphql/use-with-review-answer-state.tsx

```diff
@@ -187,17 +187,21 @@ export function useWithReviewAnswerState(
 
   function recordAnswers(status: Status, subject: string, category: string) {
     navigate(
-      `/record?back=${encodeURI(
-        `/?subject=${selectedSubject}`
-      )}&status=${status}&subject=${subject}&category=${category}`
+      urlBuild("/record", {
+        status: status,
+        subject: subject,
+        category: category,
+        back: encodeURI(`/?subject=${selectedSubject || ""}`),
```

Comment:
if there's no `selectedSubject`, would it not be better to exclude the param altogether?

```typescript
back: encodeURI(urlBuild('/', selectedSubject? { subject: selectedSubject } : {}),
```